crates/table/src/layout.rs

```diff
@@ -424,13 +425,17 @@ impl From<SumType> for SumTypeLayout {
             .collect::<Vec<_>>()
             .into();
 
-        let tag_offset = max_child_size as u16;
-        let size = align_to(max_child_size + 1, max_child_align) as u16;
-        let align = max_child_align as u16;
+        // guarantees tag fits inside align
+        let align = u16::max(max_child_align as u16, 1);
+
+        // [tag | pad to align | payload]
+        // payload may be empty
+        let size = align + align_to(max_child_size, max_child_align) as u16;
```

Comment:
```suggestion
        // Ensure the payload field is sufficiently aligned for all its members.
        // `max_child_size` and `max_child_align` will already be consistent
        // if the most-aligned variant is also the largest,
        // but this is not necessarily the case.
        // E.g. if variant A is a product of 31 `u8`s, and variant B is a single `u64`,
        // `max_child_size` will be 31 and `max_child_align` will be 8.
        // Note that `payload_size` may be 0.
        let payload_size = align_to(max_child_size, max_child_align);

        // [tag | pad to align | payload]
        let size = align + payload_size as u16;
```

I had to do some thinking to convince myself this `align_to` was necessary, so I'm suggesting we add that reasoning as a comment.

Reply:
Yeah good point, I had to stare at it for a while as well. 👍